jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostRabitRegressionSuite.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ Copyright (c) 2014 - 2019 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala.spark
+
+import ml.dmlc.xgboost4j.java.{Rabit, XGBoostError}
+import ml.dmlc.xgboost4j.scala.{Booster, DMatrix}
+
+import scala.collection.JavaConverters._
+import org.apache.spark.sql._
+import org.scalatest.FunSuite
+
+class XGBoostRabitRegressionSuite extends FunSuite with PerTest {
+  override def sparkSessionBuilder: SparkSession.Builder = super.sparkSessionBuilder
+    .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+    .config("spark.kryo.classesToRegister", classOf[Booster].getName)
+
+  test("test parity classification prediction") {
+    val training = buildDataFrame(Classification.train)
+    val testDF = buildDataFrame(Classification.test)
+
+    val model1 = new XGBoostClassifier(Map("eta" -> "1", "max_depth" -> "2", "verbosity" -> "1",
+      "objective" -> "binary:logistic", "num_round" -> 5, "num_workers" -> numWorkers)
+    ).fit(training)
+    val prediction1 = model1.transform(testDF).select("prediction").collect()
+
+    val model2 = new XGBoostClassifier(Map("eta" -> "1", "max_depth" -> "2", "verbosity" -> "1",
+      "objective" -> "binary:logistic", "num_round" -> 5, "num_workers" -> numWorkers,
+      "rabit_bootstrap_cache" -> true, "rabit_debug" -> true, "rabit_reduce_ring_mincount" -> 100,
```

Comment:
the test is too coarse-grained, 

rabitDebug should not affect test, we should not turn it on to flood out test logs or enable at most one test

but we should test rabit_bootstrap_cache (which is as the fault recovery test), rabit_reduce_buffer, and rabit_reduce_ring_mincount individually

additionally for rabit_reduce_ring_mincount, we should set it as 1 or 0 (if allowed) to guarantee that we have enabled ring topology

Reply:
rabitDebug can be removed.

rabit_bootstrap_cache failure retry only works after we path sparkcontext.stop() fail-all. It is next pr which can be another hundreds line of code. 

rabit_reduce_ring_mincount is not boolean in rabit, it's a numerical value threshold. But I agree making a boolean can help user better understand.

Reply:
removed rabit_bootstrap_cache from this pr

Reply:
I would prefer expost rabit_ring_reduce as boolean to user, if it set to true, we translate it to rabit_reduce_ring_mincount=1